src/main/java/seedu/address/model/status/Status.java

```diff
@@ -0,0 +1,89 @@
+package seedu.address.model.status;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents a Status in the address book.
+ * Guarantees: immutable; name is valid as declared in {@link #isValidStatus(String)}
+ */
+public class Status {
+
+    public static final String MESSAGE_CONSTRAINTS = "The statuses are only none, non-urgent and urgent. Please use "
+            + "only one of them";
+
+    public final StatusEnum status;
+
+    /**
+     * Constructs a {@code Status}.
+     *
+     * @param statusString A valid status value, which is found in the {@code StatusEnum}
+     */
+    public Status(String statusString) {
+        requireNonNull(statusString);
+        checkArgument(isValidStatus(statusString), MESSAGE_CONSTRAINTS);
+        if (statusString.isEmpty()) {
+            this.status = StatusEnum.NONE;
+        } else {
+            this.status = StatusEnum.valueOf(statusString.toUpperCase());
+        }
+    }
+
+    /**
+     * Returns true if a given string is a valid status value.
+     *
+     * @param test The string to be tested
+     */
```

Comment:
Would be better to have `@return` tag in the Javadoc comment and explaination of what boolean return value means :D

Reply:
Hi I have added this, thank you pleng.